src/ecephys_etl/modules/vbn_create_stimulus_table/create_stim_table.py

```diff
@@ -0,0 +1,551 @@
+            f"is not guaranteed to be correct!"
+        )
+
+    falling_edges = sync_dataset.get_falling_edges(vsync_line, units='seconds')
+
+    return falling_edges
+
+
+def get_stim_starts_and_ends(
+    sync_dataset: Dataset, fallback_line: Union[int, str] = 5
+) -> Tuple[np.ndarray, np.ndarray]:
+    """Get stimulus presentation start and end times from a loaded session
+    *.sync datset.
+
+    Parameters
+    ----------
+    sync_dataset : Dataset
+        A loaded *.sync file for a session (contains events from
+        different data streams logged on a global time basis)
+    fallback_line : Union[int, str], optional
+        The sync dataset line label to use if named line labels could not
+        be found, by default 5.
+
+        For more details about line labels see:
+        https://alleninstitute.sharepoint.com/:x:/s/Instrumentation/ES2bi1xJ3E9NupX-zQeXTlYBS2mVVySycfbCQhsD_jPMUw?e=Z9jCwH  # noqa: E501
+
+    Returns
+    -------
+    Tuple[np.ndarray, np.ndarray]
+        A tuple of numpy arrays containing
+        (stimulus_start_times, stimulus_end_times) in seconds.
+    """
+
+    # Look for 'stim_running' line in sync dataset line labels
+    stim_line: Union[int, str] = fallback_line
+    for line in sync_dataset.line_labels:
+        if line == 'stim_running':
+            stim_line = line
+            break
+        if line == 'sweep':
+            stim_line = line
+            break
+
+    if stim_line == fallback_line:
+        logging.warning(
+            f"Could not find 'stim_running' nor 'sweep' line labels in "
+            f"sync dataset ({sync_dataset.dfile.filename}). Defaulting to "
+            f"using fallback line label index ({fallback_line}) which "
+            f"is not guaranteed to be correct!"
+        )
+
+    # 'stim_running'/'sweep' line is high while visual stimulus is being
+    # displayed and low otherwise
+    stim_starts = sync_dataset.get_rising_edges(stim_line, units='seconds')
+    stim_ends = sync_dataset.get_falling_edges(stim_line, units='seconds')
+
+    return stim_starts, stim_ends
+
+
+def get_frame_offsets(
+    sync_dataset: Dataset, frame_counts: List[int], tolerance: int = 0
+) -> List[int]:
+    """List of the inferred start frames for each of the stimuli
+
+    Parameters
+    ----------
+    sync_dataset : Dataset
+        Sync data from an ecephys session (a 'Dataset' object that has loaded
+        a *.sync H5 file).
+    frame_counts : List[int]
+        List of expected frame counts (taken from behavior, mapping, and replay
+        pkl files) for each stimuli category. The list of counts should be
+        ordered by the actual display sequence of the stimuli categories.
+    tolerance : int
+        Percent by which frame counts are allowed to deviate from,
+        by default 0.
+
+    Returns
+    -------
+    List[float]
+        List of the inferred start frames for each stimuli category presented
```

Comment:
only one branch of the conditional is "inferred"

Reply:
I've created a comment in this PR to track this, will likely need feedback from stakeholder:
https://github.com/AllenInstitute/ecephys_etl_pipelines/pull/3#discussion_r640050349

I'm going to mark this particular comment as resolved as further discussion should go into the linked comment.